### DISCUSSION TITLE
Response handling

### DISCUSSION THREAD

**Discussion by Bugzy-can** [nextgenhealthcare/connect#4593] [Q&A]
I have an HL7 (in/out) channel, with a TCP listener Source, and HTTP Sender destination. 
In my sender transformer, I check the inbound HL7 message, and if there is a problem, I have it set to throw "Error". 
This flags it as an error and I can review it later, but at the same time, Mirth tosses a generic (automatic) reply, that has nothing in MSA-3. 
I want to copy the error used in the throw, into that field when it goes out, but, when I issue a throw, the message never gets to the destination response, so everything I try to do in there fails.
I can create two global map variables, one to indicate there is an error, and the second to hold the error string, and then not use throw, in which case the destination response code is used, but I am not sure how to modify the field to add the value on outbound, or if this is even possible.
I am sure there will be a simple way to manage this, if someone can enlighten me.

**Comment by pacmano1** [comment]
You likely have Mirth set to automatically generate an ACK.  In your case you need to create your own ACK.    You can find example over on the mirth forums site.

**Comment by michaeljeffreycollins** [comment]
pacmano1 most likely has the right answer. But to confirm it we need to see your actual channel setup. Response selection, data types, templates selected to confirm.   Most likely switch the source queue off, then respond from Destination X will fix this.
